peru/plugin.py

```diff
@@ -99,11 +148,11 @@ async def _plugin_job(plugin_context, module_type, module_fields, command, env,
 
         try:
             await create_subprocess_with_handle(
-                shell_command_line,
+                plugin_command,
                 display_handle,
                 cwd=plugin_context.cwd,
                 env=complete_env,
-                shell=True)
+                shell=is_shell_mode)
```

Comment:
It looks like we previously always shelled out (enabled shell mode), but this change conditionally enables shell mode exclusively on Windows (when not using `sys.executable` with Python plugins). Is that okay? (Seems fine to me, but I wanted to highlight this since there may be some subtle behavioral changes when using `create_subprocess_exec`. :shrug:)

Reply:
Yeah previously I guess I thought we might as well do it the same way everywhere for consistency. We could keep doing it everywhere now too. But I realized while I was refactoring this code, that `list2cmdline` was only ever intended to match the behavior of the Windows shell, and it's kind of a coincidence that it produces output that's been working for us on Unix, and there might be some cases where it breaks things that we don't know about. (Also the source module for that function includes the comment "NOTE: We intentionally exclude list2cmdline as it is considered an internal implementation detail.")

You're right though: In an internal-implementation-details sort of sense, it is a big change to how things work. It's possible that it breaks something surprising. But for the most part, if all our tests pass on Linux/Windows/macOS, I'm pretty confident we're ok here. And if I think about weird hypothetical breaks, it's a lot easier for me to come up with those hypotheticals in shell mode than it is for me to come up with them if we don't use the shell. So maybe this will be a win for less weird breakage overall?

Reply:
> ... it's a lot easier for me to come up with those hypotheticals in shell mode ... So maybe this will be a win for less weird breakage overall?

Yeah, this is probably the right way to do this. LGTM.